models/src/node/mod.rs

```diff
@@ -221,42 +221,46 @@ impl BottlerocketNodeStatus {
     }
 
     pub fn target_version(&self) -> Version {
-        // TODO This could panic if a custom `brn` is created with improperly specified versions; however, we are removing this
+        // TODO This could panic if a custom `brs` is created with improperly specified versions; however, we are removing this
         // attribute in an impending iteration, so we won't fix it.
         Version::from_str(&self.target_version).unwrap()
     }
 }
 
-/// Indicates the specific k8s node that BottlerocketNode object is associated with.
+/// Indicates the specific k8s node that BottlerocketShadow object is associated with.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct BottlerocketNodeSelector {
+pub struct BottlerocketShadowSelector {
     pub node_name: String,
     pub node_uid: String,
 }
 
-impl BottlerocketNodeSelector {
-    pub fn from_bottlerocket_node(brn: &BottlerocketNode) -> error::Result<Self> {
-        let node_owner = brn
+impl BottlerocketShadowSelector {
+    pub fn from_bottlerocket_shadow(brs: &BottlerocketShadow) -> error::Result<Self> {
+        let node_owner = brs
             .metadata
             .owner_references
             .as_ref()
-            .ok_or(Error::MissingOwnerReference { brn: brn.clone() })?
+            .ok_or(Error::MissingOwnerReference { brs: brs.clone() })?
             .first()
-            .ok_or(Error::MissingOwnerReference { brn: brn.clone() })?;
+            .ok_or(Error::MissingOwnerReference { brs: brs.clone() })?;
 
-        Ok(BottlerocketNodeSelector {
+        Ok(BottlerocketShadowSelector {
             node_name: node_owner.name.clone(),
             node_uid: node_owner.uid.clone(),
         })
     }
 
-    pub fn brn_resource_name(&self) -> String {
-        format!("brn-{}", self.node_name)
+    pub fn brs_resource_name(&self) -> String {
+        format!("brs-{}", self.node_name)
     }
 }
 
-impl fmt::Display for BottlerocketNodeSelector {
+impl fmt::Display for BottlerocketShadowSelector {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "({}, {})", self.node_name, self.node_uid)
     }
 }
+
+pub fn brs_name_from_node_name(node_name: &str) -> String {
```

Comment:
I’m not sure if this method meet the expectation that @cbgbt  mentioned something that the compiler will catch. But otherwise, we have to generate `selector`  first with `node_name` and `uid` , which I think much more complex. : )